mysys/crc32/crc32_x86.c

```diff
@@ -55,7 +57,7 @@ typedef uint8_t byte;
 
 # define _gcry_bswap32 __builtin_bswap32
 
-#if __GNUC__ >= 4 && defined(__x86_64__) && defined(HAVE_CLMUL_INSTRUCTION)
+#if __GNUC__ >= 4 && defined(__x86_64__)
```

Comment:
Why do we need this `#if` at all?

I could accept that we need
```c
#ifdef __x86_64__
```
until someone cleans up the code so that it works on IA-32, but why the `__GNUC__` check? The preceding line is defining something `__GNUC__` specific.

Reply:
again, original code had it this way so we continued with it.
#if defined(_GCRY_GCC_VERSION) && _GCRY_GCC_VERSION >= 40400 /* 4.4 */
/* Prevent compiler from issuing SSE instructions between asm blocks. */

pragma GCC target("no-sse")
#endif

If this doesn't hold true any more then we can get rid of it.
Please suggest. Wondering how all these refactor comment codes made in at first place?